src/coreclr/src/inc/corjit.h

```diff
@@ -251,6 +251,14 @@ class ICorJitInfo : public ICorDynamicInfo
         UINT32 ExecutionCount;
     };
 
+    struct ClassProfile
+    {
+        enum { SIZE = 8, SAMPLE_INTERVAL = 32 };
```

Comment:
I'll add some description.

For count overflow I currently don't have great answers; it should be rare (we'd need methods that spent a lot of cycles in Tier0) but obviously it's not impossible. Right now overflow won't cause any crashes but it might lead to nonsensical profile data.